### ISSUE TITLE
testsuite: intermittent failure in "wreckrun: --error supported"

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#569]
The following test fails intermittently under travis-ci:

```
expecting success: 
	flux wreckrun --output=test2.out --error=test2.err \
	    sh -c "echo >&2 this is stderr; echo this is stdout" &&
        $WAITFILE --timeout=1 -p "this is stderr" test2.err &&
        $WAITFILE --timeout=1 -p "this is stdout" test2.out
this is stderr
this is stdout
waitfile: wakeup
waitfile: Timeout after 1s
-rw-rw-r-- 1 travis travis 15 Feb 17 18:06 test2.err
not ok 37 - wreckrun: --error supported
```

Based on `waitfile --timeout` behavior, this test timed out waiting for pattern to appear in `test2.err`, even though stdout/err appears to indicate that the correct output was found. This could be a bug in `waitfile.lua` script...

**Comment by garlick** [comment]
This travis-ci failure looks like it could be the same failure, in a different subtest?
```
expecting success: 
	flux wreckrun --output=test1.out echo hello &&
        $WAITFILE --timeout=1 --pattern=hello test1.out
hello
waitfile: wakeup
waitfile: Timeout after 1s
-rw-rw-r-- 1 travis travis 6 Feb 22 17:48 test1.out
not ok 36 - wreckrun: --output supported
```

**Comment by grondo** [comment]
Yeah, I have a suspicion this is a problem with the `waitfile.lua` script. Let me push a patch to run `waitfile.lua` verbosely. Are you hitting this regularly when re-running travis builds?

**Comment by garlick** [comment]
I think I've hit it two or three times in maybe a dozen (x4) runs.

**Comment by dongahn** [comment]
@grondo: Thank you for looking into this. If there is indeed a bug in `waitfile.lua` and you happen to fix, it would be good to see if the intermittent error captured in #512 still occurs with the fix.

**Comment by dongahn** [comment]
Also FYI -- I added [`waitfile.lua`](https://github.com/flux-framework/flux-sched/blob/master/t/scripts/waitfile.lua) into the flux-sched sometime back, so if a patch is made, we need to apply that to the scheduler as well.

**Comment by grondo** [comment]
OK. Thanks.. I'm trying to reproduce in travis now with some debugging enabled.

**Comment by garlick** [comment]
I just hit it [here](https://travis-ci.org/flux-framework/flux-core/jobs/111048970):
```
expecting success: 
	flux wreckrun --output=test2.out --error=test2.err \
	    sh -c "echo >&2 this is stderr; echo this is stdout" &&
        $WAITFILE -v --timeout=1 -p "this is stderr" test2.err &&
        $WAITFILE -v --timeout=1 -p "this is stdout" test2.out
this is stderr
this is stdout
waitfile: File appeared with size 0
waitfile: wakeup
waitfile: back to sleep
waitfile: Timeout after 1s
-rw-rw-r-- 1 travis travis 15 Feb 22 20:20 test2.err
not ok 37 - wreckrun: --error supported
```

**Comment by grondo** [comment]
Thanks! Hm, might need to add a bit more debug to see what's happening between the internal "wakeup" and "back to sleep". The logic to determine if there was a change in the file might be wrong. Sorry!

**Comment by grondo** [comment]
Hm, rookie mistake? comparing mtime in seconds:
I'm not sure the file modification check here should even be necessary since `waitfile.lua` is now using statwatcher. I think it was holdover from the version that did essentially `sleep(1) && check()`

```lua
    if st.st_mtime > prev.st_mtime then
        log_verbose ("File changed\n")
        if st.st_size < prev.st_size then
            printf ("truncated!\n")
            self.position = 0 -- reread
        end
        return true
    end
```

**Comment by grondo** [comment]
Since this has not reproduced in awhile, assuming fixed by #580
